### ISSUE TITLE
can I  set a default D-I file  so when I  install the ubuntu 22.04 server ,I do not need to select options and user name ,etc.

### ISSUE THREAD

**Issue by gitgonewithwind** [PJ-Singh-001/Cubic#223]
Can I  set a default D-I file  so when I  install the ubuntu 22.04 server ,I do not need to select options and user name ,etc ?
so that I can install everyting do not need human-computer interaction？
I watched the video here: https://www.youtube.com/watch?v=vjXLkqu893Y
but it only can add some packages in ISO.
I read here: https://help.ubuntu.com/community/InstallCDCustomization
but it seems not support ubuntu 22.04 server.

Can Cubic do it like that? 
if possible,how to do ? where is the guide

Thank you

**Comment by PJ-Singh-001** [comment]
See if this information helps:

 1. How to use cloud-int (a.k.a. Automated Server installation, a.k.a. autoinstall config) in Cubic:

    https://github.com/PJ-Singh-001/Cubic/issues/101#issuecomment-1336728758

 2. Here is the reference for autoinstall:

    https://ubuntu.com/server/docs/install/autoinstall-reference

 3. Here is the schema for the autoinstall file:

    https://ubuntu.com/server/docs/install/autoinstall-schema

**Comment by gitgonewithwind** [comment]
@PJ-Singh-001  thanks. let me try

**Comment by gitgonewithwind** [comment]
@PJ-Singh-001 
I completely followed  your link https://github.com/PJ-Singh-001/Cubic/issues/101#issuecomment-1336728758
but can not install successfully
error output:
waiting for clould-init...
finish:subiquity/Mirror/load_autoinstall_data
Aerror occurred, please enter to start a shell.
after click "enter" into command line.

when I modified boot/grub/grub.cfg and boot/grub/loopback.cfg, I opened the files, there was no "boot=casper" in my default files. anyway, I completely copy  boot=casper autoinstall "ds=nocloud;s=/cdrom/preseed/"   to add it as below:

I am using host linux lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal


and cubic 
cubic --version
Cubic version... 2023.05.83
Copyright....... © 2015, 2020 PJ Singh
Website......... https://github.com/PJ-Singh-001/Cubic



any help?
Thanks.
I am going to delete "boot=casper" to try

I tried to remove  "boot=casper" and only add  autoinstall "ds=nocloud;s=/cdrom/preseed/" , but the same error happened


Add autoinstall "ds=nocloud;s=/cdrom/preseed/" as shown:

menuentry "Try or Install Ubuntu Server" {
    set gfxpayload=keep
    linux	/casper/vmlinuz boot=casper autoinstall "ds=nocloud;s=/cdrom/preseed/"  ---
    initrd	/casper/initrd.gz
}
boot/grub/grub.cfg
boot/grub/loopback.cfg

Add autoinstall "ds=nocloud;s=/cdrom/preseed/" as shown:

menuentry "Try or Install Ubuntu Server" {
    set gfxpayload=keep
    linux	/casper/vmlinuz boot=casper autoinstall "ds=nocloud;s=/cdrom/preseed/"  ---
    initrd	/casper/initrd.gz
}

**Comment by gitgonewithwind** [comment]
@PJ-Singh-001 
anyway, I only need 1. lvm,encrypt my disk with passphrase  2 user: myuser, password and all other option I can keep default in installation.
it seems not hard to do(before I deleted  your preseed other users and only keep one user and remove the last identified input user. when I installed , only show disk storage and input user. others options never show, it seemed work partly. let me try more
